python/TestHarness/TestHarness.py

```diff
@@ -46,10 +47,23 @@ def findTestRoot(start=os.getcwd(), method=os.environ.get('METHOD', 'opt')):
 class TestHarness:
 
     @staticmethod
-    def buildAndRun(argv, app_name, moose_dir):
+    def buildAndRun(argv, app_name, moose_dir, unit=None, unit_dir=None, unit_run_make=True):
         harness = TestHarness(argv, moose_dir, app_name=app_name)
         harness.findAndRunTests()
-        sys.exit(harness.error_code)
+        unit = bool(os.getenv('MOOSE_RUN_TESTS_WITH_UNIT', False)) if unit is None else unit
```

Comment:
I'm not sure I'm sold on this either. @WilkAndy is right, we should just be doing this already. The unit tests are always very fast. If we can just run them with the tests that would be best. I guess I'm OK with a variable to control them, but maybe we should be defaulting this on, not off?

This is all good, but I think we should separate building and running. Let's look into building the unit app when we build the application.

Reply:
That is fine with me, I am just don't know anything about make.

Reply:
The reason it is not on by default is because all of our civet recipes have separate unit test sections. If you change make then we can remove all the unit test parts from civet.

Reply:
Thanks for working on this - I do think it's in the right direction. We might even consider incorporating MOOSEDocs into parts of this (not per unit test, but maybe at some aggregate level). I was really worried that the assessors where going to say something about the complete lack of requirements/design/testing in our unit tests. Perhaps they just didn't know enough to ask 😉

Reply:
@permcody Are you going to take a crack at the Makefile? If so, just use this branch having moose/test/unit make all the framework, modules, and apps consistent.

Reply:
requirements/design/testing in unit tests: i'm not 100% convinced this is a good idea.  The unit tests are mostly for developers to check the fundamentals of their code and are never(?) for user-centric functionality.